src/Hosting/Hosting/src/StaticWebAssets/StaticWebAssetsFileProvider.cs

```diff
@@ -40,20 +44,29 @@ public StaticWebAssetsFileProvider(string pathPrefix, string contentRoot)
         /// <inheritdoc />
         public IDirectoryContents GetDirectoryContents(string subpath)
         {
-            if (!StartsWithBasePath(subpath, out var physicalPath))
+            var modifiedSub = subpath.StartsWith("/") ? subpath : "/" + subpath;
```

Comment:
This needs unit tests. Add unit tests for the new behavior here https://github.com/aspnet/AspNetCore/blob/master/src/Hosting/Hosting/test/StaticWebAssets/StaticWebAssetsFileProviderTests.cs